bring-it-up/bring-it-up
# [FE#13] Create filter bar component

## Description
#### Summary:
- Adds a filter bar component to the home page with 5 filters: Support Type, Delivery Method, Urgency, Specialty, and School

#### Changes:
- Adds reusable `FilterDropdown.tsx` component for a single filter category
- Adds `FilterBar.tsx` component that consists of 5 filters and a button to clear the filters
- Modifies the search bar and home page CSS so that it's structured the same as the wireframe
- Adds 2 extra data fetches in the home page to retrieve the list of schools and list of specialties
- Makes Urgency and School as single-select filters (radio buttons)
- Makes Support Type, Delivery Method, and Specialty as multi-select filters (check boxes)

#### How to test this PR:
- Visit the Netlify deploy preview link below (data might take a while to load the first time if the backend server is down and needs to restart. Could be a couple of minutes!)
- After data has loaded, all 5 filters should have a list of options (might need to refresh)
- Try different combos of filters, the services should reload as soon as you change an option
  - The filtering works as follows: It returns services that match the requirement of EVERY filter. If a filter is multi-select, then it returns services that match ANY of the selected options. This logic has already been tested on the backend, so we just need to make sure we are sending the correct query string.
  - You can verify it's working by using the browser debug tools and looking at the network requests sent. The request payload should show the query parameters that correspond to the filter (see screenshots for examples)
  - The "Any" option in a single-select filter doesn't include a query parameter in the request
- Clicking "Clear All Filters" should show all the data again 

## Screenshots:
<img width="1440" alt="Screenshot 2023-04-04 at 6 34 26 PM" src="https://user-images.githubusercontent.com/57027339/229958570-d80ab9e3-f9f9-403a-8d80-2acb138b5fcb.png">
<img width="1440" alt="Screenshot 2023-04-04 at 6 36 31 PM" src="https://user-images.githubusercontent.com/57027339/229958814-caf6fc37-6814-4879-9d32-98dd62c3fd18.png">
  
## Checklist
